src/node/app.ts

```diff
@@ -22,40 +22,37 @@ export interface App extends Disposable {
   server: http.Server
 }
 
-const listen = (server: http.Server, { host, port, socket, "socket-mode": mode }: ListenOptions) => {
-  return new Promise<void>(async (resolve, reject) => {
+export const listen = async (server: http.Server, { host, port, socket, "socket-mode": mode }: ListenOptions) => {
+  if (socket) {
+    try {
+      await fs.unlink(socket)
+    } catch (error: any) {
+      handleArgsSocketCatchError(error)
+    }
+  }
+  await new Promise<void>(async (resolve, reject) => {
     server.on("error", reject)
-
     const onListen = () => {
       // Promise resolved earlier so this is an unrelated error.
       server.off("error", reject)
       server.on("error", (err) => util.logError(logger, "http server error", err))
-
-      if (socket && mode) {
-        fs.chmod(socket, mode)
-          .then(resolve)
-          .catch((err) => {
-            util.logError(logger, "socket chmod", err)
-            reject(err)
-          })
-      } else {
-        resolve()
-      }
+      resolve()
     }
-
     if (socket) {
-      try {
-        await fs.unlink(socket)
-      } catch (error: any) {
-        handleArgsSocketCatchError(error)
-      }
-
       server.listen(socket, onListen)
     } else {
       // [] is the correct format when using :: but Node errors with them.
       server.listen(port, host.replace(/^\[|\]$/g, ""), onListen)
     }
   })
+
+  // NOTE@jsjoeio: we need to chmod after the server is finished
+  // listening. Otherwise, teh socket may not have been created yet.
+  if (socket && mode) {
+    await fs.chmod(socket, mode).catch((err) => {
+      util.logError(logger, "socket chmod", err)
+    })
```

Comment:
Sorry for not catching this earlier, we should let this error throw so code-server will exit:

```suggestion
    await fs.chmod(socket, mode)
```

Reply:
Nice last-minute-typical-super-asher-catch! 🔥

Reply:
You:
![image](https://user-images.githubusercontent.com/3806031/157956449-a69c9335-6f8d-4294-ae79-078146ee34d7.png)

Reply:
woah how did you find a photo of me, I thought that was impossible

Reply:
Wayback Machine. Found it on your old freelancing portfolio site

Reply:
hahahahahaha :joy: